admin_guide/managing_networking.adoc

```diff
@@ -1147,3 +1147,50 @@ objects:
       - podSelector: {}
 ...
 ----
+
+[[admin-guide-enabling-hsts]]
+== Enabling HTTP Strict Transport Security
+
+HTTP Strict Transport Security (HSTS) policy is a security enhancement, which
+ensures traffic to a host will always be HTTPS. This is useful for ensuring
+secure interactions with websites, or to offer secure application for the user's
+benefit.
+
+HSTS adds a Strict Transport Security header to HTTPS responses from the site.
+The client (often a browser) reads the header and from then on enforces the
+policy. All requests to the HTTP URL are changed to HTTPS before the request is
+sent, eliminating the need for a redirect.
+
+When not enabled, any HTTP requests to a host that require HTTPS instead
+results in the host generating a redirect to the HTTPS URL, or drops the
+request.
+
+To enable HSTS to a route, add the `haproxy.router.openshift.io/hsts_header`
```

Comment:
This only works on secure routes (edge terminated or re-encrypt). Adding the annotation to http or passthrough doesn't help.